### PR TITLE
Edge to Edge new api

### PR DESCRIPTION
**Pull request by android-dev-lxl**

### CODE REVIEW

**Review by johnshea — APPROVED** [review]
lgtm

File: app/src/main/java/com/example/superheroes/ui/theme/Theme.kt [hunk 1]
```diff
@@ -117,7 +115,9 @@ fun SuperheroesTheme(
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
-            setUpEdgeToEdge(view, darkTheme)
+            val window = (view.context as Activity).window
+            window.statusBarColor = colorScheme.primary.toArgb()
```
Comment (Jaehwa-Noh) [inline app/src/main/java/com/example/superheroes/ui/theme/Theme.kt:119]:
```suggestion
            window.statusBarColor = Color.Transparent.toArgb()
```
You need to make status bar color transparency.

File: app/src/main/java/com/example/superheroes/ui/theme/Theme.kt [hunk 2]
```diff
@@ -117,7 +115,9 @@ fun SuperheroesTheme(
...
+            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
```
Comment (Jaehwa-Noh) [inline app/src/main/java/com/example/superheroes/ui/theme/Theme.kt:120]:
```suggestion
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !darkTheme
```
Need to not operation at darkTheme to show the status information: time, battery, etc.

**Review by Jaehwa-Noh — CHANGES_REQUESTED** [review]
Hello, I'd found your PR as study new edge-to-edge, and I suggested some changes at your code to work properly.

File: app/src/main/java/com/example/superheroes/MainActivity.kt [hunk 3]
```diff
@@ -28,13 +28,15 @@ import androidx.compose.material3.Surface
...
 import androidx.compose.ui.Modifier
+import androidx.activity.enableEdgeToEdge
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import com.example.superheroes.model.HeroesRepository
 import com.example.superheroes.ui.theme.SuperheroesTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
+        enableEdgeToEdge()
```
Comment (Jaehwa-Noh) [inline app/src/main/java/com/example/superheroes/MainActivity.kt:39]: According this [documentation](https://developer.android.com/jetpack/compose/layouts/insets#compose-apis) 
I think that enableEdgeToEdge() should be placed after super.onCreate(savedInstanceState)

스크린샷 2023-11-25 091112

**Review by yenerm — APPROVED** [review]

**Review by Jaehwa-Noh — APPROVED** [review]
Thank you to accept my suggestion.
Look good to me!

**Pull request merged**